### ISSUE TITLE
Inappropriate destination while using "Goto definition" option in Decompiler Mode

### ISSUE THREAD

**Issue by MizzMaster** [Col-E/Recaf#389]
While I was working on an obfuscated jar, I found a bug that causes you to find yourself in a random class that is not even relevant to what you would expect. Let me introduce a few images to explain.

I was on a pretty clean class and I though there was no problem to use "Goto definiton" option.
![image](https://user-images.githubusercontent.com/69322376/119928040-71744b80-bf83-11eb-93cb-1767c1beae45.png)

 I used it and... I was here...
![image](https://user-images.githubusercontent.com/69322376/119928145-ada7ac00-bf83-11eb-8371-20e29c3d392c.png)
I was expecting a Packet class but I found a Predicate, my mind just blowed first time I faced this bug

Fortunately, I found a way to find the actual class after I let go of my persistent thoughts about a "clean class". I just opened it in the assembler.
![image](https://user-images.githubusercontent.com/69322376/119929321-f2ccdd80-bf85-11eb-981c-0dcccaab0672.png)
It was the actual class. But it was in the **default package**. This is where the problem starts...
As Java do not allow importing classes in **default package** from a non-default package class and the other way around (since Java 1.5), we can't import like that. So, CFR couldn't put the import statement of that class. But Recaf somehow recognizes this class as a **same-package** class I guess (because there is no import statement for that) and redirects me to the **same-package**d class. (Both Predicate class named **j1** and it's initializer class are in the `net.minecraft` package)

**Comment by Col-E** [comment]
Looks like the JavaParser resolve logic is failing in this case.

**Comment by MizzMaster** [comment]
https://workupload.com/file/h4wWFDDQuek

![image](https://user-images.githubusercontent.com/69322376/119946295-25cf9b00-bf9f-11eb-91ce-363199331a2e.png)

**Comment by Col-E** [comment]
Yeah, validated my prior assumption. Because the context is based on the decompiled code and not the underlying bytecode it sees that we're in `X` package and pulls the matching class name from the package we're in, despite it pointing to the class by the same name in the default package _(Which can't be done at source level)_.

I'd have to incorporate some double layer context system to address this since the information needed to resolve this simply cannot be represented in decompiled code.

**Comment by MizzMaster** [comment]
> There's no good way to resolve bytecode instruction correlation to decompiled source code in any of the decompilers AFAIK

There might be, but for now you may can prevent this bug by simply checking the bytecode instructions if there is another class exist with the same **Simple Name**. (Compare what Recaf tries to redirect and what bytecode includes)

I'm going to explain here what I'm trying to say. (cuz my english is bad)
As in my example jar, when you right click the **SampleClass.class** or **SampleClass.test()**, before Recaf pops up the options, Recaf should check all the bytecode instructions (of **UnCreativeClassName.randomMethod**) that contains a class name (`LDC`, `INVOKESTATIC`, etc.). If there is another class with exact **Simple Name** comparing to what Recaf found in the decompiler mode, then Recaf should warn the user somehow. 

(**Goto Definition** option might become strikethrough text and might show a text box while hovering over that says something like "Destination may not be true, check the bytecode in order to be sure")

**Comment by Col-E** [comment]
That's the double layer approach, but warning instead of resolving.
